platform/linux-generic/odp_pool.c

```diff
@@ -718,36 +713,47 @@ int buffer_alloc_multi(pool_t *pool, odp_buffer_hdr_t *buf_hdr[], int max_num)
 		buf_hdr[i] = buf_hdr_from_index(pool, cache->buf_index[j]);
 	}
 
-	/* Declare variable here to fix clang compilation bug */
-	uint32_t data[burst];
-
 	/* If needed, get more from the global pool */
 	if (odp_unlikely(num_deq)) {
 		/* Temporary copy to data[] needed since odp_buffer_t is
 		 * uintptr_t and not uint32_t. */
+		uint32_t data[CACHE_BURST];
+		uint32_t burst = 0;
+		uint32_t deq = 0;
+		uint32_t mask;
+		ring_t *ring;
+
 		ring      = &pool->ring->hdr;
 		mask      = pool->ring_mask;
-		burst     = ring_deq_multi(ring, mask, data, burst);
-		cache_num = burst - num_deq;
 
-		if (odp_unlikely(burst < num_deq)) {
-			num_deq   = burst;
-			cache_num = 0;
-		}
+		while (num_deq) {
+			burst = ring_deq_multi(ring, mask, data, CACHE_BURST);
```

Comment:
Why do you need the loop here and just don't dequeue all the required buffers at once? Clang bug?

 ring_deq_multi() is the expensive call here (especially when multiple processes are stressing the pool), so I don't like calling it unnecessarily.